tests/integ/modin/frame/test_cache_result.py

```diff
@@ -31,66 +41,76 @@ def assert_empty_snowpark_pandas_equals_to_pandas(snow_df, native_df):
         assert native_snow_df.index.equals(native_df.index)
 
 
+@pytest.fixture(scope="function")
+def simple_test_data():
+    return {
+        "col0": ["foo", "bar", "baz", "foo", "bar", "baz", "foo", "bar", "baz"],
+        "col1": ["abc", "def", "ghi", "ghi", "abc", "def", "def", "ghi", "abc"],
+        "col2": list(range(9)),
+    }
+
+
 def cache_and_return_df(snow_df, inplace):
     """
     Helper method to cache and return a reference to cached DataFrame depending on inplace.
+
+    Notes
+    -----
+    If inplace=True, the returned df is a reference to the inputted df.
     """
     if not inplace:
         cached_snow_df = snow_df.cache_result()
     else:
         # If inplace=True, there is no return, so we set `cached_snow_df` equal to `snow_df`.
-        snow_df.cache_result(inplace=inplace)
         cached_snow_df = snow_df
+        cached_snow_df.cache_result(inplace=inplace)
     return cached_snow_df
 
 
+def perform_chained_operations(df, module):
+    df = df.reset_index(drop=True)
```

Comment:
add a comment that we expect this to be expensive because each `concat` unions results, and each result may itself contain unions?